frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -576,7 +617,37 @@ function ReaderHighlight:_resetHoldTimer(clear)
     end
 end
 
+function ReaderHighlight:onTogglePanelZoomSetting(arg, ges)
+    if not self.document.info.has_pages then return end
+    self.panel_zoom_enabled = not self.panel_zoom_enabled
+    self:onSaveSettings()
```

Comment:
I believe this onSaveSettings is unnecessary (it will be called when reader is closed and will save the updated self.panel_zoom_enabled ).